[WT-75]: Feature/upload folder

larry-internxt opened the pull request:
- Added uploadFolder from htmlinput logic
- Added uploadFolder button and popover

larryrider committed:
added uploadFolder from htmlinput logic

larryrider committed:
moved folder service

larryrider committed:
added uploadFolder button and popover

larryrider committed:
removed console.log

fabioespinosa commented on src/app/drive/services/folder.service/uploadFolderInput.service.ts:
```diff
@@ -0,0 +1,64 @@
+    for (const file of files) {
+        file.webkitRelativePath.split('/').reduce(
+            (previousValue, currentValue, currentIndex, arrayPaths) => {
+                //console.log(currentIndex, previousValue, currentValue);
```
remove this comment

fabioespinosa commented on src/app/drive/services/folder.service/uploadFolderInput.service.ts:
```diff
@@ -0,0 +1,64 @@
+        file.webkitRelativePath.split('/').reduce(
+            (previousValue, currentValue, currentIndex, arrayPaths) => {
+                //console.log(currentIndex, previousValue, currentValue);
+                if (currentIndex == arrayPaths.length - 1) {
```
use always triple equals ===

fabioespinosa commented on src/app/drive/services/folder.service/uploadFolderInput.service.ts:
```diff
@@ -0,0 +1,64 @@
+        }
+    }
+    return childrenFolders;
+};
```
Add a line after every file for end of file

larryrider committed:
removed outline and fixed warning from popover button

larryrider committed:
added some fixes on uploadFolderInput

aboredvaro committed:
Small UI fixes

- Z-index problem with new folder dialog fixed.
- Icon sizes, spacing and weights fixed.

larryrider committed:
changed uploadFolder's component from popover to dropdown menu

larryrider committed:
refactored menuitems

larryrider committed:
redesign and some fixes on createFolderDialog

larryrider committed:
refactored renaming files logic and added buttons to actionbar

larryrider committed:
changed breadcrumbs logic and design

larryrider committed:
fixed renaming files cursor position and selectedItems not being updated

fabioespinosa commented on src/app/drive/components/CreateFolderDialog/CreateFolderDialog.tsx:
```diff
@@ -47,29 +63,38 @@ const CreateFolderDialog = ({ onFolderCreated, currentFolderId }: CreateFolderDi
     <BaseDialog isOpen={isOpen} title="Create folder" onClose={onClose}>
-      <div className="w-64 self-center mt-4">
+      <div className="mt-4 px-5">
+        <span className={'text-sm'}>Name</span>
```
className="text-sm"

fabioespinosa commented on src/app/drive/components/DriveExplorer/DriveExplorer.tsx:
```diff
@@ -62,255 +83,356 @@ interface DriveExplorerProps {
-  get hasItems(): boolean {
-    return this.props.items.length > 0;
-  }
+  const hasAnyItemSelected = (): boolean => {
```
Since this.props.items changes on every render you don't have to declare them as functions but as values like:

`const hasItems = this.props.items.length >0;`

Same for `hasAnyItemSelected`, `hasItems` and `hasFilters`

fabioespinosa commented on src/app/shared/components/Breadcrumbs/Breadcrumbs.tsx:
```diff
@@ -7,26 +9,70 @@ interface BreadcrumbsProps {
+    if (hiddenItemsList.length > 0) {
+      const menu = <Dropdown
+        key={'breadcrumbDropdownItems'}
+        classButton={
```
it can be a string here instead of {}

fabioespinosa commented on src/app/shared/components/Breadcrumbs/Breadcrumbs.tsx:
```diff
@@ -7,26 +9,70 @@ interface BreadcrumbsProps {
+        classButton={
+          'mx-1 flex items-center justify-center transition-all duration-75 ease-in-out'
+        }
+        openDirection={'left'}
```
here as well

fabioespinosa commented on src/app/shared/components/Breadcrumbs/Breadcrumbs.tsx:
```diff
@@ -7,26 +9,70 @@ interface BreadcrumbsProps {
+          'mx-1 flex items-center justify-center transition-all duration-75 ease-in-out'
+        }
+        openDirection={'left'}
+        classMenuItems={'left-1 mt-10 w-max rounded-md border border-black border-opacity-8 bg-white py-1.5 drop-shadow'}
```
and here

fabioespinosa reviewed (changes requested):
Correct the changes and it should be good to go 👌

larryrider committed:
some code fixes

larryrider committed:
added title to span to show text on long namefiles hover

larryrider committed:
fixed breadcrumbs css bug

larryrider committed:
changed breadcrumb title color

larryrider committed:
Merge branch 'master' into feature/upload-folder-input

larryrider committed:
removed database polling every 1.5 secs

aboredvaro committed:
Fixed UI issues

- Breadcrumb overflow menu
- Skin skeleton

larryrider committed:
added parent folder to breadcrumb

fabioespinosa committed:
fix(foler-dialod): change BaseDialog for Modal

fabioespinosa committed:
fix(DriveExplorer): double conditional for CreateFolderDialog

aboredvaro committed:
Fix Delete and CreateFolder dialogs

aboredvaro committed:
Fixed Breadcrumb
